src/client/models/tankShot.js

```diff
@@ -0,0 +1,867 @@
+        var sprite = new Image();
+        sprite.src = './public/images/explosion_sheet.png';
+        function tick(coords){
+            var xExplosion = coords.x - 40;   // x = x-central - R;
+            var yExplosion = coords.y - 40;   // y = y-central - R;
+
+            clear();
+
+            fillBackground(); // it's instead of ctx.clearRect(0, 0, canvas.width, canvas.height);
+            drawTank(tankX, tankY);
+            ctx.drawImage(sprite, xSprite, 0, 134, 134, xExplosion, yExplosion, 134, 134);
+            if (xSprite < 1608) {
+                xSprite = xSprite + 134;
+                window.setTimeout(tick, 70, coords);
+                // console.log('Coords are: ' + coords.x + ' and ' + coords.y);
+            } else {
+                xSprite = 0;
+            }
+        }
+
+
+        //<------Maks's part-------->
+
+
+        var requestAnimFrame = (function(){
+            return window.requestAnimationFrame   ||
+            window.webkitRequestAnimationFrame ||
+            window.mozRequestAnimationFrame    ||
+            window.oRequestAnimationFrame      ||
+            window.msRequestAnimationFrame     ||
+            function(callback){
+                window.setTimeout(callback, 1000 / 60);
+            };
+        })();
+        var power =  parseInt(getId('power').innerHTML);
+        var angle = parseInt(getId('angle').innerHTML);;
+        var lastTime;
+        var dt2=0;
+        var bullets = [];
+        var bullet;
+        var lastFire = Date.now();
+        var gameTime = 0;
+        var bulletImg=new Image();
+        bulletImg.src='./public/images/bullet2.png';
+
+        function drawBullet() {
+            clear();
+
+            fillBackground();
+            drawTank(tankX,tankY);
+
+            var now = Date.now();
+            var dt = (now - lastTime) / 1000.0;
+            update(dt);
+            render();
+            lastTime = now;
+        }
+
+        function shotStart() {
+            lastTime = Date.now();
+            drawBullet();
+        }
+
+        function update(dt) {
+            gameTime += dt;
+            updateEntities(dt);
+        }
+
+        function updateEntities(dt) {
+            for(var i=0; i<bullets.length; i++) {
+                bullet = bullets[i];
+                bullet.pos[0] = tankX+45 + bullet.bulletSpeed * dt2*Math.cos(bullet.angle*Math.PI/180);
+                bullet.pos[1]=tankY-44-(bullet.bulletSpeed*dt2*Math.sin(bullet.angle*Math.PI/180)-9.8*dt2*dt2/2);
+                dt2+=2*dt;
+
+                var coords = {x:bullet.pos[0],
+                    y:bullet.pos[1],
+                    width:10,
+                    height:10};
+
+                if (checkCol(coords,originalPoints)){
+                    console.log( 'x:' +  (coords.x + coords.width), 'y:' + (coords.y + coords.height));
+                    bullets.splice(i, 1);
+                    tick(coords);                                   // <------ Explosion ------>
```

Comment:
May I suggest a better name for this function? Like `generateExplosion` for instance